File: Puntonet/Puntonet.Web/Modules/ServerTypes/Parameters/ColorsRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface ColorsRow {
    IdColor?: number;
    Description?: string;
}

export abstract class ColorsRow {
    static readonly idProperty = 'IdColor';
    static readonly nameProperty = 'Description';
    static readonly localTextPrefix = 'Parameters.Colors';
    static readonly lookupKey = 'Parameters.Colors';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<ColorsRow>('Parameters.Colors') }
    static async getLookupAsync() { return getLookupAsync<ColorsRow>('Parameters.Colors') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<ColorsRow>();
}

File: Puntonet/Puntonet.Web/Modules/Parameters/Colors/ColorsRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace Puntonet.Parameters
{
    [ConnectionKey("Default"), Module("Parameters"), TableName("[dbo].[Colors]")]
    [DisplayName("Colors"), InstanceName("Colors")]
    [ReadPermission("Administration:General")]
    [ModifyPermission("Administration:General")]
    public sealed class ColorsRow : Row<ColorsRow.RowFields>, IIdRow, INameRow
    {
        [DisplayName("Id Color"), Identity, IdProperty]
        public int? IdColor
        {
            get => fields.IdColor[this];
            set => fields.IdColor[this] = value;
        }

        [DisplayName("Description"), Size(200), NotNull, QuickSearch, NameProperty]
        public string Description
        {
            get => fields.Description[this];
            set => fields.Description[this] = value;
        }

        public ColorsRow()
            : base()
        {
        }

        public ColorsRow(RowFields fields)
            : base(fields)
        {
        }

        public class RowFields : RowFieldsBase
        {
            public Int32Field IdColor;
            public StringField Description;
        }
    }
}